core/src/main/scala/cats/syntax/either.scala

```diff
@@ -189,9 +254,13 @@ final class EitherOps[A, B](val eab: Either[A, B]) extends AnyVal {
    * }}}
    */
   def toEitherT[F[_]: Applicative]: EitherT[F, A, B] = EitherT.fromEither(eab)
+
+  private[cats] def leftCast[C]: Either[C, B] = eab.asInstanceOf[Either[C, B]]
+
+  private[cats] def rightCast[C]: Either[A, C] = eab.asInstanceOf[Either[A, C]]
```

Comment:
I wonder if it's worth exposing these actually. As syntax enrichments over `Left` and `Right`.